packages/querygrid/src/components/user/ChangePasswordModal.tsx

```diff
@@ -0,0 +1,121 @@
+import * as React from 'react'
+import { Button, Col, Form, FormControl, Modal, Row } from "react-bootstrap";
+import { Alert, LabelHelpTip, User } from "@glass/base";
+import { changePassword, getPasswordRuleInfo } from "./actions";
+import { ChangePasswordModel } from "./models";
+
+interface Props {
+    user: User
+    onSuccess: () => any
+    onHide: () => any
+}
+
+interface State {
+    model: ChangePasswordModel
+    passwordRule: string
+    submitting: boolean
+    error: string
+}
+
+export class ChangePasswordModal extends React.Component<Props, State> {
+
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            model: new ChangePasswordModel({
+                email: props.user.email
+            }),
+            passwordRule: undefined,
+            submitting: false,
+            error: undefined
+        };
+    }
+
+    componentDidMount() {
+        getPasswordRuleInfo()
+            .then((response) => {
+                this.setState(() => ({passwordRule: response.summary}));
+            });
+    }
+
+    onChange = (evt) => {
+        const name = evt.target.id;
+        const value = evt.target.value;
+
+        this.setState((state) => ({
+            model: state.model.set(name, value) as ChangePasswordModel
+        }));
+    };
+
+    submitChangePassword = () => {
+        this.setState(() => ({submitting: true}));
+
+        changePassword(this.state.model)
+            .then((result) => {
+                this.props.onHide();
+                this.props.onSuccess();
```

Comment:
Do  you want to set submitting to false here?

Reply:
Done.